src/netlink.c

```diff
@@ -0,0 +1,362 @@
+	        return 2;
+	}
+        lua_pushnumber(L, 1);
+        return 1;
+}
+
+
+/*-------------------------------------------------------------------------*\
+* Closes socket used by object
+\*-------------------------------------------------------------------------*/
+static int meth_close(lua_State *L)
+{
+    p_netlink nl = (p_netlink) auxiliar_checkgroup(L, "netlink{any}", 1);
+    socket_destroy(&nl->fd);
+    lua_pushnumber(L, 1);
+    return 1;
+}
+
+/*-------------------------------------------------------------------------*\
+* Just call tm methods
+\*-------------------------------------------------------------------------*/
+static int meth_setpeername(lua_State *L){
+	p_netlink nl;
+	struct sockaddr_nl addr;
+	if(lua_isnone(L,2)){
+		nl=auxiliar_checkclass(L,"netlink{connected}",1);
+		addr.nl_family=AF_UNSPEC;
+		socket_connect(&nl->fd,(struct sockaddr *)&addr,(socklen_t)sizeof(addr),&nl->tm);
+		auxiliar_setclass(L,"netlink{unconnected}",1);
+		return 0;
+	}
+	nl=auxiliar_checkclass(L,"netlink{unconnected}",1);
+	int dstpid = luaL_checkinteger(L,2);
+	int grps = luaL_optinteger(L,3,0);
+	addr.nl_pid=dstpid;
+	addr.nl_groups=grps;
+	addr.nl_family=AF_NETLINK;
+
+	int err=socket_connect(&nl->fd,(struct sockaddr *)&addr,(socklen_t)sizeof(addr),&nl->tm);
+	if(err!=IO_DONE){
+		lua_pushnil(L);
+		lua_pushstring(L,socket_strerror(errno));
+		return 2;
+	}
+	auxiliar_setclass(L,"netlink{connected}",1);
+	lua_pushinteger(L,err);
+	return 1;
+}
+
+static int meth_getpeername(lua_State *L){
+	p_netlink nl = (p_netlink) auxiliar_checkclass(L,"netlink{connected}",1);
+	struct sockaddr_nl peer;
+	socklen_t peer_len = sizeof(peer);
+	if (getpeername(nl->fd, (SA *) &peer, &peer_len) < 0) {
+		lua_pushnil(L);
+		lua_pushstring(L, socket_strerror(errno));
+	    	return 2;
+	}
+	lua_pushinteger(L,peer.nl_groups);
+	lua_pushinteger(L,peer.nl_pid);
+	return 2;
+}
+/*-------------------------------------------------------------------------*\
+* Just call tm methods
+\*-------------------------------------------------------------------------*/
+static int meth_settimeout(lua_State *L)
+{
+    p_netlink nl = (p_netlink) auxiliar_checkgroup(L, "netlink{any}", 1);
+    return timeout_meth_settimeout(L, &nl->tm);
+}
+
+static int meth_gettimeout(lua_State *L)
+{
+    p_netlink nl = (p_netlink) auxiliar_checkgroup(L, "netlink{any}", 1);
+    return timeout_meth_gettimeout(L, &nl->tm);
+}
+
+/*=========================================================================*\
+* Library functions
+\*=========================================================================*/
+/*-------------------------------------------------------------------------*\
+* Creates a master netlink object
+\*-------------------------------------------------------------------------*/
+static int global_create(lua_State *L)
+{
+    t_socket sock;
+    int err = socket_create(&sock, AF_NETLINK, SOCK_RAW, NETLINK_USERSOCK);
+    /* try to allocate a system socket */
```

Comment:
Too many comments, strive for code so clean that comments aren't needed. This comments actually disturb the reading.

Reply:
If it was 100% my API I'd agree with you, but I'm following the standard set by Diego, so I can hopefully send him a pull request eventually